### PR TITLE
ENIFP-3285: Allow Modal Header Size Alteration

### PR DESCRIPTION
**Pull request by bradyclifford**
Currently the size of the header cannot be changed.

### CODE REVIEW

**Comment by bradyclifford** [comment]
I am sure I am missing some things in this PR.  For now this is just a proposal.

**Review by stevematney — CHANGES_REQUESTED** [review]
This change is fine, but I would like us to handle the prop types slightly differently.

Thank you!

File: packages/es-components/src/components/containers/modal/ModalHeader.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import styled from 'styled-components';
 import OrigDismissButton from '../../controls/DismissButton';
 import { ModalContext } from './ModalContext';
 import Heading from '../heading/Heading';
-import { HeadingLevel } from 'es-components-shared-types';
+import { HeadingLevel, HeadingSize } from 'es-components-shared-types';
```
[the remark this answers is not in the document]
Comment (bradyclifford) [reply packages/es-components/src/components/containers/modal/ModalHeader.tsx]: This PR was just a pseudo example of what we wanted.

[the remark this answers is not in the document]
Comment (bradyclifford) [reply packages/es-components/src/components/containers/modal/ModalHeader.tsx]: > "es-components-shared-types"' has no exported member named 'headingLevel'. Did you mean 'HeadingLevel'?

I will probably just have to pull this down to my local box and compile it.  Up until this point, I was only using the GitHub UI.

**Comment by bradyclifford** [comment]
Any idea why the Cypress tests are failing?  Its failing on components this PR does not even touch.

**Comment by bradyclifford** [comment]
> ## Commitlint
> commitlint results
> 
> The following commit messages do not meet our commit convention. Please reword these commits for correct versioning and changelog generation.
> 
> * Allow Modal Header Size Alteration
>   
>   * subject may not be empty
>   * type may not be empty
> * Update packages/es-components/src/components/containers/modal/ModalHeader.tsx
> 
> Co-authored-by: Steve Matney [steve.matney@extendhealth.com](mailto:steve.matney@extendhealth.com) * subject may not be empty * type may not be empty
> 
> * Update packages/es-components/src/components/containers/modal/ModalHeader.tsx
> 
> Co-authored-by: Steve Matney [steve.matney@extendhealth.com](mailto:steve.matney@extendhealth.com) * subject may not be empty * type may not be empty
> 
> * Update packages/es-components/src/components/containers/modal/ModalHeader.tsx
> 
> Co-authored-by: Steve Matney [steve.matney@extendhealth.com](mailto:steve.matney@extendhealth.com) * subject may not be empty * type may not be empty
> 
> * New: Allow modal header level alteration
>   
>   * subject must not be sentence-case, start-case, pascal-case, upper-case

Every commit has to have that format?

**Comment by stevematney** [comment]
> Every commit has to have that format?
@bradyclifford yes.

**Comment by stevematney** [comment]
> Any idea why the Cypress tests are failing? Its failing on components this PR does not even touch.

It's failing because `HeadingLevel` is only a type and not a js variable. That's why I recommended [this change](https://github.com/WTW-IM/es-components/pull/808#discussion_r1458017298).

https://github.com/WTW-IM/es-components/blob/cf2556f5b045f10043a5bb23e9e54250ebbe2573/shared/types/src/ESTheme.ts#L206

**Comment by bradyclifford** [comment]
> > Any idea why the Cypress tests are failing? Its failing on components this PR does not even touch.
> 
> It's failing because `HeadingLevel` is only a type and not a js variable, so it can't build. That's why I recommended [this change](https://github.com/WTW-IM/es-components/pull/808#discussion_r1458017298).
> 
> https://github.com/WTW-IM/es-components/blob/cf2556f5b045f10043a5bb23e9e54250ebbe2573/shared/types/src/ESTheme.ts#L206

Oh ya that makes sense.  I must of missed that.  What I get for trying to make these adjustments from GitHub itself.

**Review by stevematney — APPROVED** [review]

**Pull request merged**